framework/src/modules/interoperability/base_state_recovery.ts

```diff
@@ -130,16 +108,50 @@ export class BaseStateRecoveryCommand<
 		const {
 			params: { chainID, storeEntries, module, siblingHashes },
 		} = context;
-		const storeQueries = [];
+		const storeQueriesVerify = [];
+		const queryKeys = [];
+		const storePrefix = computeStorePrefix(module);
+
+		for (const entry of storeEntries) {
+			const queryKey = Buffer.concat([
+				storePrefix,
+				entry.substorePrefix,
+				utils.hash(entry.storeKey),
+			]);
+			queryKeys.push(queryKey);
+			storeQueriesVerify.push({
+				key: queryKey,
+				value: utils.hash(entry.storeValue),
+				bitmap: entry.bitmap,
+			});
+		}
+		const terminatedStateAccount = await this.stores
+			.get(TerminatedStateStore)
+			.get(context, chainID);
+
+		const sparseMerkleTree = new SparseMerkleTree();
+		const proofOfInclusionStores = { siblingHashes, queries: storeQueriesVerify };
+		// The SMT verification step is computationally expensive. Therefore,
+		// it is done in the execution step such that the transaction fee must be paid.
+		const smtVerified = await sparseMerkleTree.verify(
```

Comment:
Unrelated to this PR, but why is `verify` a method of the SMT class? Couldn't it be just an independent function as specified in the LIP0039, so that we don't need to instantiate a new SMT instance?

Reply:
Related PR: https://github.com/LiskHQ/lips/pull/432

Reply:
the reason we have it is coz its coming from Lisk DB and in order to access it we need instantiate the SMT class